### DISCUSSION TITLE
Do already installed applications get moved?

### DISCUSSION THREAD

**Discussion by hardbrocklife** [ValShaped/rwfus#47] [Q&A] [answered]
I understand conceptually what this does, essentially superimposing a intermediary root file system on top of the actual rootfs to avoid losing pacman installed applications.

What about applications you already have installed? Do they get moved or updated to using overlayFS?

**Comment by ValShaped** [comment, marked as answer]
No, unfortunately not. Rwfus is not very smart, and operates entirely by mounting and unmounting filesystems. It has a few safeguards against making your system *entirely* unbootable, but it doesn't integrate with `pacman` or read package lists. For a bash script, it's already far too complicated for its own good 😅

I originally built it to install a small number of self-contained dev tools (i.e. Yakuake) on my Steam Deck without unlocking the root filesystem, and for that purpose, it works pretty well.

**Comment by hardbrocklife** [comment]
No worries. Thanks for the response. I appreciate your work.

Would you advise against installing it now, and do so after a SteamOS update, or does it not matter? Im fairly proficient with Linux, as I can fix a lot of the package and build issues that occur pretty much every time manjaro updates lol. I just don't want to mindlessly bork something because its a new concept to me.

**Reply by ValShaped** [reply]
I'd recommend upgrading first, so pacman doesn't attempt to upgrade any dependencies alongside the package you install. Dependencies are the real hurdle here, since Rwfus will effectively freeze them at whatever version you install.

**Comment by hardbrocklife** [comment]
Thank you! I appreciate you taking the time to respond to a rando on the interest. May
the Sun bless you and keep you.
